libs/datasets/timeseries.py

```diff
@@ -782,6 +782,39 @@ def add_new_cases(timeseries: MultiRegionTimeseriesDataset) -> MultiRegionTimese
     return new_timeseries
 
 
+def _calculate_modified_zscore(series: pd.Series, window: int = 10) -> pd.Series:
+    """Calculates zscore removing zeros"""
```

Comment:
I think it's worth documenting the implementation for those that don't deal with pandas abstractions often. Some of the edge cases feel like unwanted behavior.

Some things to include:
- Trailing equally weighted window based only on valid measurements
- 0 is not treated as a valid measurement (add why?)
- Needs at least 3 valid (non-NaN, non-Zero) measurements in the window to return any value
- Time shifts the values forward one day (I don't understand why this is yet, feels like an off by one error but maybe I haven't found the right index alignments)
- No minimum positive value of cases needed for thresholding (pd.Series([1,1,1,2,1]) triggers right now, which feels wrong, I think that's just a 1/0 -> inf shifted to the right one but I don't know)

Given the number of different people that will want to know, and our push towards transparency in how we handle missing data, I think this needs to be much more explicit.

Reply:
If the time shift by 1 is to get the trailing sample mean exclusive of the new candidate point, then I'd prefer we just made that window shape more explicit.